egs/callhome_diarization/v1/local/VB_resegmentation.py

```diff
@@ -0,0 +1,308 @@
+            print("Warning: {} has no voiced frames in the initialization file".format(utt))
+            continue
+
+        # Initialize the posterior of each speaker based on the clustering result.
+        if args.initialize:
+            q = VB_diarization.frame_labels2posterior_mx(init_ref_voiced, args.max_speakers)
+            if args.true_rttm_filename != "None": 
+                cmd = "md-eval.pl -1 -c 0.25 -r {}/{}_true.rttm -s {}/{}.rttm 2".format(temp_dir, utt, temp_dir, utt)
+                status, output = commands.getstatusoutput(cmd) 
+                assert status == 0
+                DER_info = match_DER(output)
+                print("BEFORE RUNNING VB RESEGMENTATION")
+                print(DER_info + "\n")
+        else:
+            q = None
+            print("RANDOM INITIALIZATION\n")
+        
+        # VB resegmentation
+
+        # q  - S x T matrix of posteriors attribution each frame to one of S possible
+        #      speakers, where S is given by opts.maxSpeakers
+        # sp - S dimensional column vector of ML learned speaker priors. Ideally, these
+        #      should allow to estimate # of speaker in the utterance as the
+        #      probabilities of the redundant speaker should converge to zero.
+        # Li - values of auxiliary function (and DER and frame cross-entropy between q
+        #      and reference if 'ref' is provided) over iterations.
+        q_out, sp_out, L_out = VB_diarization.VB_diarization(X_voiced, m, iE, w, V, sp=None, q=q, maxSpeakers=args.max_speakers, maxIters=args.max_iters, VtiEV=None,
+                                  downsample=args.downsample, alphaQInit=args.alphaQInit, sparsityThr=args.sparsityThr, epsilon=args.epsilon, minDur=args.minDur,
+                                  loopProb=args.loopProb, statScale=args.statScale, llScale=args.llScale, ref=None, plot=False)
+
+        predicted_label_voiced = np.argmax(q_out, 1) + 2
+        predicted_label = (np.zeros(len(mask))).astype(int)
+        predicted_label[mask] = predicted_label_voiced
+
+        duration_list = []
+        for i in range(args.max_speakers):
+            num_frames = np.sum(predicted_label == (i + 2))
+            if num_frames == 0:
+                continue
+            else:
+                duration_list.append(1.0 * num_frames / len(predicted_label))
+        duration_list.sort()
+        duration_list = map(lambda x: '{0:.2f}'.format(x), duration_list)
+        print("PREDICTED {} SPEAKERS".format(len(duration_list)))
+        print("DISTRIBUTION {}".format(" ".join(duration_list)))
+        print("sp_out", sp_out)
+        print("L_out", L_out)
+
+        # Create the output rttm file and compute the DER after re-segmentation
+        create_rttm_output(utt, predicted_label, rttm_dir, args.channel)
+        if args.true_rttm_filename != "None":
+            cmd = "md-eval.pl -1 -c 0.25 -r {}/{}_true.rttm -s {}/{}_predict.rttm 2".format(temp_dir, utt, rttm_dir, utt)
```

Comment:
In my opinion this script should produce an RTTM file. The evaluation of it should happen outside of this script, in the top-level run.sh.